### ISSUE TITLE
Error: Error occurred in token received callback function. Error: Invalid state. <guid>, state expected : null..

### ISSUE THREAD

**Issue by m4ss1m0g** [AzureAD/microsoft-authentication-library-for-js#1468]
## Library
- [X] `msal@1.2.2` and `msal@1.3.0-beta.0`
- [ ] `@azure/msal-browser@2.x.x`
- [ ] `@azure/msal-angular@0.x.x`
- [ ] `@azure/msal-angular@1.x.x`
- [ ] `@azure/msal-angularjs@1.x.x`

## Framework
- @vue/cli@4.3.0
- Vue 2.6.11
- Vuex 3.1.3
- Vuetify: 2.2.20

## Description
All tests made 
- `In Private`
- On compiled code

After the `loginRedirect` with successfully authentication **ONLY With Firefox 74.0.1 x64** calling the `handlerRedirectCallback` raise the Error on the subject.

If I add the code mentioned on #960 the error gone for Firefox

``` js
   setTimeout(function() {
      instance.handleRedirectCallback(...)
  }, 500);
```
But break ALL other browsers

I has also tried the `msal@1.3.0-beta.0` mentioned on #1229 with no luck

## Configuration
``` js

var GLOBAL_MSAL_CONFIG = {
    name: "development",
    configuration: {
        auth: {
            clientId: "<guid>",
            authority: "https://login.microsoftonline.com/<guid>",
            redirectUri: "http://localhost:8888"
        },
        cache: {
            cacheLocation: "localStorage",
            storeAuthStateInCookie: false
        }
    },
    request: {
        scopes: ["api://<guid>/access_as_user"]
    }
};

```

## Reproduction steps
1. Run the attached code into NGINX
2. Open the Firefox, click the LOGIN button and authenticate
3. When redirect to the page on console there are an error

## Expected behavior
Successfully authenticate

## Browsers
- Firefox 74.0.1 x64 

## Attach
[Manifest.zip](https://github.com/AzureAD/microsoft-authentication-library-for-js/files/4440867/Manifest.zip)
[myapp.nginx.conf.zip](https://github.com/AzureAD/microsoft-authentication-library-for-js/files/4440868/myapp.nginx.conf.zip)
[MyApp.zip](https://github.com/AzureAD/microsoft-authentication-library-for-js/files/4440869/MyApp.zip)

**Comment by ryandegruyter** [comment]
We are experiencing this issue with a select few users.
Session state is not persisted resulting in a mismatch.

"@azure/msal-angular": "0.1.4",

Microsoft Edge

Any idea's what the cause might be?

**Comment by m4ss1m0g** [comment]
I notice two things:
1.  On debug (vue serve) it working properly with Firefox but when transpiled with webpack the problem occurs. Maybe a Time problem ? When debugging is slightly more slow than compiled.
If I add a `setTimeout` it working.
2. Occurs with Firefox `InPrivate` window

I try to debug and if I have more clues I drop a comment.

Thanks.

**Comment by jasonnutter** [comment]
@m4ss1m0g Can you please try `msal@1.3.0-beta.1` and let us know if the issue still occurs?

**Comment by tnorling** [comment]
@m4ss1m0g We expect that this may have been fixed by `msal@1.3.0`. Let us know if that is not the case.
